plutus-core/plutus-core/src/PlutusCore/Builtins.hs

```diff
@@ -219,6 +221,10 @@ instance (GShow uni, GEq uni, DefaultUni <: uni) => ToBuiltinMeaning uni Default
         makeBuiltinMeaning
             ((++) :: String -> String -> String)
             mempty  -- TODO: budget.
+    toBuiltinMeaning EqualsString =
+        makeBuiltinMeaning
+            ((==) @String)
+            mempty  -- TODO: budget.
```

Comment:
I was going to question this, but then I see we have the same for the other string builtins, sigh.

Reply:
Are strings "official" builtins that we expect to be used in real contracts?  I was aware that the costing functions for these were missing, but I was under the impression that strings weren't supposed to be for general use, only debugging (they've never been mentioned in the PLC specification, for example).  It'd be easy enough to add costing, but (a) it'd increase the time required by the already lengthy builtin-benchmarking process, and (b) I don't think we have any programs that make extensive use of strings.

Reply:
... or is this working towards the "add more bytestring builtins" thing, which I suppose we will need to cost properly?

Reply:
I think we need to include them.
- The node will have a verbose output mode that will include the log strings, so they will actually get used on-chain.
- It's fairly impractical to not include them on-chain, due to the code changes you'd need and the fact that it'd change all the script hashes...

Reply:
I've mad an issue to add costing for the string builtins: https://jira.iohk.io/browse/SCP-2315. I don't think it's massively urgent, but we should remember to do it.

Reply:
I think we can have a single `Equals` builtin that works for every built-in type (as long as ``uni `Everywhere` Eq`` ofc). Do we care?

Reply:
It's definitely going to be non-trivial to assign a cost for it, though.

Reply:
Seems like an argument for not doing it :)

Reply:
(actually, there would be another problem: we can provide a Haskell denotation, but how would the type of such a builtin look on the PLC side? `equals : all a. a -> a -> bool` is a lie as it's only supposed to work on constants. So it'd have to be `equals : all a. sing a -> a -> a -> bool`, but we don't have that quite yet)